torch/nn/modules/module.py

```diff
@@ -1368,11 +1368,15 @@ def state_dict(self, *, prefix: str = ..., keep_vars: bool = ...) -> Dict[str, A
     # TODO: Change `*args` to `*` and remove the copprespinding warning in docs when BC allows.
     # Also remove the logic for arg parsing together.
     def state_dict(self, *args, destination=None, prefix='', keep_vars=False):
-        r"""Returns a dictionary containing a whole state of the module.
+        r"""Returns a dictionary containing references to a whole state of the module.
 
         Both parameters and persistent buffers (e.g. running averages) are
         included. Keys are corresponding parameter and buffer names.
         Parameters and buffers set to ``None`` are not included.
+        
+        .. warning::
```

Comment:
maybe `.. note::` instead of `.. warning::` ? I don't think there's anything inherently dangerous about returning references

Reply:
Say you have a list of `state_dicts`, one per epoch, or have `best_state = net.state_dict() if val_loss < best_val_loss else best_state`… You want these to be different objects / copies.

Reply:
[Others](https://ofstack.com/python/40554/solve-the-state_dict-of-copy-problem-of-pytorch.html) have been bitten by this already.

Reply:
the equivalent of `state_dict` generally in Python is `__getstate__`. `__getstate__` is a shallow-copy. It is first deep-copied before it is used in other contexts.

with this context, it feels like leaning towards a note than a warning. what do you think?

Reply:
Alright. Better a note than nothing, I guess 😛